piano-hero/mobile/src/main/java/com/novoda/pianohero/GameMvp.java

```diff
@@ -0,0 +1,36 @@
+package com.novoda.pianohero;
+
+interface GameMvp {
+
+    interface Model {
+
+        void startGame(StartCallback callback);
+
+        RoundViewModel onNotesPlayed(CompletionCallback callback, Note... note);
+
+        interface StartCallback {
+            void onGameStarted(RoundViewModel viewModel);
+        }
+
+        interface CompletionCallback {
+            void onGameComplete();
+        }
+    }
+
+    interface View {
+
+        void showRound(RoundViewModel viewModel);
+
+        void showGameComplete();
+    }
+
+    interface Presenter {
+
+        void onCreate();
+
+        void onNotesPlayed(Note... notes);
```

Comment:
similarly (though it could be argued we're shortcutting), neither the view nor model should know about the presenter. In this example, the View knows the presenter.

should these methods just be **separate** callbacks that the view knows about (/owns), _provided_ by the presenter?